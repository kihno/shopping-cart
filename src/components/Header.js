import { Link } from 'react-router-dom';
import React from 'react';
import CartButton from './CartButton';

const Header = (props) => {
    const { cart, handleCartClick } = props;

    return(
        <div id="header">
            <div id="logo">LOGO</div>
            <nav>
                <ul>
                    <Link to="/">
                        <li>Home</li>
                    </Link>
                    <Link to="/shop">
                        <li>Shop</li>
                    </Link>
                </ul>
            </nav>
            <CartButton cart={cart} handleCartClick={handleCartClick} />
        </div>
    )
}

export default Header;