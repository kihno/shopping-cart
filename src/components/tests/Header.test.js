import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../Header';
import { BrowserRouter } from 'react-router-dom';

const cart = [];

describe('Header', () => {
    it ('renders header', () => {
        const { container } = render(<BrowserRouter><Header cart={cart} /></BrowserRouter>);
        expect(container).toMatchSnapshot();
    });
});