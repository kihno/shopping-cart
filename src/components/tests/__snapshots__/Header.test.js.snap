// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header 1`] = `
<div>
  <div
    id="header"
  >
    <div
      id="logo"
    >
      LOGO
    </div>
    <nav>
      <ul>
        <a
          href="/"
        >
          <li>
            Home
          </li>
        </a>
        <a
          href="/shop"
        >
          <li>
            Shop
          </li>
        </a>
      </ul>
    </nav>
    <div
      id="cartButton"
    >
      <img
        alt="Shopping Cart"
        id="shoppingCart"
        src="shoppingBag.svg"
      />
      <span
        data-testid="cartTotal"
        id="cartTotal"
      />
    </div>
  </div>
</div>
`;
